### ISSUE TITLE
Using edit and continue will crash visual studio

### ISSUE THREAD

**Issue by jltjohanlindqvist** [dotnet/roslyn#12656]
When pausing a running process (x64) in debug mode and editing code visual studio will crash when trying to save. This only happens in our server project, but it is consistent.

**Version Used**: Visual Studio Professional 2015 Version 14.0.25424.00 Update 3

**Steps to Reproduce**:
1. Start process
2. Pause process
3. Edit code
4. Save
5. Visual studio will crash when trying to save

**Expected Behavior**:
The changed code will be saved and process resumes

**Actual Behavior**:
Save is completed but the client will crash (I can say the save is completed because when I restart visual studio the edited code was saved)

**Comment by tmat** [comment]
Could you please save and share a memory dump of the VS process (devenv.exe) using ```procdump -ma devenv``` command line?

**Comment by jltjohanlindqvist** [comment]
I have a dump file from when I recorded the bug using the report a problem function in Visual Studio, will that one suffice or is there something missing from it that you'd rather I make one with procdump?

**Comment by tmat** [comment]
Let me take a look. It might be sufficient.

**Comment by jltjohanlindqvist** [comment]
Have you had some time to look into this? Also if the dump file was sufficient or if I need to use procdump?

**Comment by tmat** [comment]
@jltjohanlindqvist Sorry for the delay. I believe this is the same issue as https://github.com/dotnet/roslyn/issues/9497

**Comment by jltjohanlindqvist** [comment]
Seems like you're correct. Tried using edit and continue in a smaller method and it worked there, but when I tried it again in a large method it crashed. Looking at the Task Manager when Visual Studio is about to crash shows the devenv instance allocating memory up to like 2.2 GB and then dropping down to 600-700 MB a bunch of times.

Is this a problem that came with Roslyn? Because afaik another developer at the company uses Visual Studio 2013 and doesn't have any issues using edit and continue in large methods.
